### ISSUE TITLE
Failed to execute 'observe' on redirect in gatsby

### ISSUE THREAD

**Issue by kamilsnopkowski** [alvarotrigo/react-fullpage#325]
### Description

Hi, I am a contributor in project made on gatsby-starter-default setup and I'm experiencing an issue while redirecting through the site. The idea is to have single page using fullPage and after i redirect (using gatsby-router) to another one I get an error:
`Failed to execute 'observe' on 'MutationObserver': parameter 1 is not of type 'Node'.`

I believe it is caused because other pages do not have any DOM structure related to the fullPage plugin, therefore I've tried to do sort of cleanup like this:
```
useLayoutEffect(() => {
  if (pageId === 'fullpage_id') {
    window.fullpage_api.reBuild();
  } else {
    window.fullpage_api.destroy('all');
  }
}
```
But the problem still occurs.

### Versions 
@fullpage/react-fullpage: 0.1.24

**Comment by alvarotrigo** [comment]
Can you provide an isolated reproduction in [Codesandbox](https://codesandbox.io/s/m34yq5q0qx) with no CSS or JS files external to fullPage.js and the minimum amount of HTML code? Use empty sections unless strictly necessary for the reproduction.

**Comment by kamilsnopkowski** [comment]
## First of all thanks for quick reply
 
For now I wasn't really able to reproduce the issue but in meantime I found something that could be a usefull information. I've noticed that `destroy('all')` method doesn't really undo the changes in the html and since Links in gatsby-router do not refresh the page, some classes/styles preserve, Here I drop screens of not fullPage related page before and after visiting the one with the plugin:
### Before
![image](https://user-images.githubusercontent.com/83779239/168148834-786c8746-b509-428f-8e16-3107ec9d93d8.png)
### After
![image](https://user-images.githubusercontent.com/83779239/168148684-ad8e8bae-05a8-4c55-9315-3fef1d84f83d.png)

Forcing site to reload with `window.location.reload()` kinda works but it's slow so the error has time to show up and well it kills the idea behind routing in react/gatsby

**Comment by alvarotrigo** [comment]
Thanks for that! 👍 
So the main issue here is that the inline styles don't get removed in:

- body
- parents of fullpage.js containers 

- As well as the "fp-scrollable" class on the `body`

Right?

**Comment by kamilsnopkowski** [comment]
Yes but I'm not 100% sure if that's the actuall reason and removing those will fix the error. In quite similiar issue I've found you were suggesting to remove classes manually so I will try that approach

**Comment by kamilsnopkowski** [comment]
Removing styles and classes on their own did not work for me, but I've noticed that on your example [here](https://codepen.io/alvarotrigo/pen/bdxBzv) class fp-scrollable does not get removed on calling destroy, It's being added. Is that the expected behaviour?

https://user-images.githubusercontent.com/83779239/168540478-29ed0af4-e753-413e-9021-8a9435cab869.mp4

**Comment by alvarotrigo** [comment]
Those classes should be removed, so that's for sure a bug.
However, that is very probably not the issue of the problem. 

The problem is probably caused by trying to initialize fullPage.js before the fullPage.js structure is ready (or when it's nonexistent).

You can reproduce the issue on the small fiddle I've created:
https://jsfiddle.net/2jsqpd01/

I guess an easy solution for this is checking for the element we want to observe before observing it.
I'll come back soon with a link to a file you can use to test this fix.

**Comment by alvarotrigo** [comment]
Can you try replacing your dist files for the ones in the dev branch?
https://github.com/alvarotrigo/react-fullpage/tree/dev/dist

Or if you prefer, you can replace them using npm by installing the dev branch:

```sh
npm install "https://github.com/alvarotrigo/react-fullpage.git#dev" --save
```

**Comment by kamilsnopkowski** [comment]
Thanks for explaning the error!

Well i've replaced files via npm but sadly it did not fix the problem

**Comment by alvarotrigo** [comment]
Try providing an isolated reproduction so I can play with the code and inspect the error.
Without a reproduction, it will be difficult to track.

**Comment by alvarotrigo** [comment]
Is the error in the fullPage.js code btw? 
Can you paste here the line?

**Comment by pleodigital** [comment]
Yes error comes from fullPage files:

![image](https://user-images.githubusercontent.com/48985121/168747642-94edcc09-8cd5-4aaf-b8d8-5e17373d2907.png)

I will try provide the reproduction later today.

**Comment by kamilsnopkowski** [comment]
Yes error comes from fullPage files:
![image](https://user-images.githubusercontent.com/83779239/168748718-e3ac63ac-e8cf-4074-aa7b-cae40dbaf4d9.png)

I will try to provide reproduction later today.

**Comment by alvarotrigo** [comment]
If you beautify the code on Chrome we can see the exact line where it happens.
![image](https://user-images.githubusercontent.com/1706326/168799124-5a888e8a-1f28-4cb2-b00f-02bd9279c243.png)

**Comment by kamilsnopkowski** [comment]
Here you go:
![image](https://user-images.githubusercontent.com/83779239/168811476-1ac51e77-877e-4931-8ac4-fa09740fe00f.png)


`Qn().observer && xe && xe.observe(v(dn)[0], ke)`

**Comment by alvarotrigo** [comment]
It doesnt look like you are using the dev files. 
But I’ll verify when i get home

**Comment by kamilsnopkowski** [comment]
Oh yes sorry my bad, here with the dev files:
![image](https://user-images.githubusercontent.com/83779239/168817285-91b3732e-4fb6-44c9-9b44-a76ce7ae3ae2.png)
`Qn().observer && xe && xe.observe(v(dn)[0], ke)`

**Comment by alvarotrigo** [comment]
Thansk.
I've made another commit on the dev branch. Can you update your files again? Uninstall and install to make sure you get the latest. You should see a `fullPage 4.0.8-test2` on the source code.

**Comment by kamilsnopkowski** [comment]
Yeah I will let you know as soon as I come home

**Comment by kamilsnopkowski** [comment]
Updated files fixed the error! Yesterday faced some npm problems so had to try today and everything worked like a charm, thanks!

**Comment by alvarotrigo** [comment]
Awesome to hear!

**Comment by alvarotrigo** [comment]
Fixed in 0.1.25! 🎉
